include/wil/windowing.h

```diff
@@ -0,0 +1,78 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_WINDOWING_INCLUDED
+#define __WIL_WINDOWING_INCLUDED
+
+#include <WinUser.h>
+
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+namespace wil
+{
+    namespace details
+    {
+        template<typename TCallback>
+        BOOL CALLBACK LambdaEnumWindowsProc(HWND data, LPARAM lParam) noexcept
+        {
+            const TCallback* pCallback = reinterpret_cast<const TCallback*>(lParam);
+            using result_t = decltype((*pCallback)(data));
+#ifdef __cpp_if_constexpr

```

Comment:
```suggestion
#if __cpp_if_constexpr
```
or
```c++
#ifdef __cpp_if_constexpr >= 201606L
```
Major nit. Most all guidance or uses I've seen test against a version (or against 0 if you're testing presence at all)

Reply:
I'm not sure what you meant @dunhor.
If we want to check against a version, then `#if __cpp_if_constexpr >= 201606L` would be right. Here I was checking whether it's defined at all, which is equivalent as I'm not relying on post 201606 behavior. 
Saying `#ifdef __cpp_if_constexpr >= 201606L` seems wrong (the >= part should be ignored)

Reply:
Feature test macros, like `__cpp_if_constexpr`, are defined to values that indicate their "level" of support. There's two possible sources of concern:

The first is if new standards expand on the functionality. Let's hypothetically say that the first version of `__cpp_if_constexpr` did not allow braces for `if constexpr` (i.e. forcing it to be a single statement). In that case, the `#ifdef` check would pass for older compilers, but fail compilation when it encounters the `{`.

The second is for pre-release experimental features. E.g. it's theoretically possible for `__cpp_if_constexpr` to be set to some pre-release constant that correlates with a draft of the proposal that uses the syntax `constexpr if` instead of `if constexpr` (just an example; idk if such a draft or implementation exists). In that case, if a user is using an older compiler with the experimental feature enabled, the `#ifdef` check will pass, but compilation will fail because the syntax is incorrect.

Now, both of these are hypothetical scenarios, so I don't think that there's _actually_ a concern here - hence the nit. I'm just advocating for performing explicit version checks as the default.

> Saying `#ifdef __cpp_if_constexpr >= 201606L` seems wrong

Woops. I meant `#if __cpp_if_constexpr >= 201606L`